samples/Mvc.FileUpload/MultipartRequestHelper.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.IO;
+using Microsoft.Net.Http.Headers;
+
+namespace Mvc.FileUpload
+{
+    public static class MultipartRequestHelper
```

Comment:
You've had to borrow quite a bit of code here and in the controller. Are there more pieces of this that should be made public?